lib/queue.js

```diff
@@ -358,25 +357,45 @@ Queue.prototype.empty = function(){
 };
 
 /**
-  Pauses the processing of this queue.
+  Pauses the processing of this queue, locally if true passed, otherwise globally.
 
-  We use an atomic RENAME operation on the wait queue. Since we have
-  blocking calls with BRPOPLPUSH on the wait queue, as long as the queue
+  For global pause, we use an atomic RENAME operation on the wait queue. Since
+  we have blocking calls with BRPOPLPUSH on the wait queue, as long as the queue
   is renamed to 'paused', no new jobs will be processed (the current ones
   will run until finalized).
 
   Adding jobs requires a LUA script to check first if the paused list exist
   and in that case it will add it there instead of the wait list.
 */
-Queue.prototype.pause = function(){
-  return pauseResume(this, true);
+Queue.prototype.pause = function(isLocal /* Optional */){
+  if (isLocal){
+    if (!this.paused){
```

Comment:
Should this be called `pausedLocal` to clarify that it's only this worker that's paused?

Also can you add documentation for this?